tasks/staking.ts

```diff
@@ -114,8 +126,11 @@ export function load() {
       const rewardsBeforeCollect = await getStakingPoolsRewardPerToken(hre);
       console.log("Rewards before sync", rewardsBeforeCollect);
 
-      const stakingRewardsDistribution = await getStakingRewardsDistribution(hre);
-      await (await stakingRewardsDistribution.collectAllRewards(0, 100)).wait();
+      const recorder = await defaultRecorder(hre);
+
+      const stakingRewardsDistribution = toRc(await getStakingRewardsDistribution(hre), recorder);
+      const numberOfStakingPools = rewardsBeforeCollect.length;
+      await stakingRewardsDistribution.record.collectAllRewards(0, numberOfStakingPools - 1);
```

Comment:
```suggestion
      await stakingRewardsDistribution.record.collectAllRewards(0, 100);
```

we have a known bug here causing the same stakingPool to be added to the list more than once